bundles/model/org.eclipse.smarthome.model.core.test/src/test/groovy/org/eclipse/smarthome/model/core/internal/folder/test/FolderObserverTest.groovy

```diff
@@ -0,0 +1,535 @@
+	private static final String MOCK_MODEL_TO_BE_REMOVED = "MockFileInModelForDeletion.java"
+
+	/** Text to be inserted in the newly created files, so the tests can run properly on all OS */
+	private final String INITIAL_FILE_CONTENT = "Initial content"
+
+	private ConfigurationAdmin configAdmin
+	private FolderObserver folderObserverService
+	private Configuration config
+	/** An instance of the ModelRepoMock class which implements the ModelRepository interface */
+	private ModelRepoMock modelRepo
+	/** An instance of the ModelParserMock class which implements the ModelParser interface */
+	private ModelParserMock modelParserMock
+	/** The default watched directory */
+	private static String defaultWatchedDir
+
+	@Before
+	void setUp() {
+		setupWatchedDirectory()
+		setUpServices()
+		config = configAdmin.getConfiguration(FOLDEROBSERVER_PID)
+		assertNotNull(config)
+	}
+
+	private void setupWatchedDirectory() {
+		/* 
+		 * The main configuration folder's path is saved in the defaultWatchedDir variable 
+		 * in order to be restored after all the tests are finished. 
+		 * For the purpose of the FolderObserverTest class a new folder is created.
+		 * Its path is set to the ConfigConstants.CONFIG_DIR_PROG_ARGUMENT property.
+		 */
+		defaultWatchedDir = System.getProperty(ConfigConstants.CONFIG_DIR_PROG_ARGUMENT)
+		File watchDir = new File(WATCHED_DIRECTORY)
+		watchDir.mkdirs()
+		System.setProperty(ConfigConstants.CONFIG_DIR_PROG_ARGUMENT, watchDir.getPath())
+		new File(EXISTING_SUBDIR_PATH).mkdirs()
+	}
+
+	void setUpServices() {
+		modelParserMock = new ModelParserMock()
+		/* 
+		 * FolderObserver calls the addModelParser() method every time a ModelParser is 
+		 * registered in the service registry.
+		 * So calling that method explicitly is not necessary. 
+		 */
+		registerService(modelParserMock)
+		modelRepo = new ModelRepoMock()
+		registerService(modelRepo)
+		folderObserverService = getService(ManagedService, FolderObserver)
+		assertThat("Folder Observer service cannot be found",folderObserverService, is(notNullValue()))
+		/*
+		 * FolderObserver is referencing to the ModelRepository service using the 
+		 * cardinality "1..1". In order to be sure it will use our mock ModelRepository object,
+		 *  it is necessary to explicitly set it.
+		 */
+		folderObserverService.setModelRepository(modelRepo)
+		configAdmin = getService(ConfigurationAdmin)
+		assertThat("Configuration Admin service cannot be found",configAdmin, is(notNullValue()))
+	}
+
+	@After
+	void tearDown() {
+		/*
+		 * The FolderObserver service have to be stopped at the end of each test
+		 * as most of the tests are covering assertions on its initialization actions
+		 */
+		folderObserverService.deactivate()
+		deleteDirectory(new File(WATCHED_DIRECTORY))
+		modelRepo.clean()
+		config.delete()
+	}
+
+	void deleteDirectory(File path) {
```

Comment:
Okay for me, but isn't the Apache Commons IO artifact already available?
There is a static method `FileUtils.deleteDirectory(File directory)`